emcc.py

```diff
@@ -1715,7 +1715,8 @@ def check_memory_setting(setting):
     if shared.Settings.WASM_BIGINT:
       shared.Settings.LEGALIZE_JS_FFI = 0
 
-    shared.Settings.GENERATE_SOURCE_MAP = shared.Settings.DEBUG_LEVEL >= 4 and not shared.Settings.SINGLE_FILE
+    if shared.Settings.SINGLE_FILE:
+      shared.Settings.GENERATE_SOURCE_MAP = 0
```

Comment:
This seems like it should above the block at line 906

Reply:
I think it would work in either place, but in terms of style I'm not sure it should be moved. The 906 location is primarily used to parse `options.*` things, while the code here is the full assortment of `shared.Settings.*` things.

I am not totally sure why there is such a distinction, to be honest, however.

Reply:
Fair enough.   If just funny that line 806 checks this option.. but 1719 sets the option.   I would have thought that we always want the sets to come before the checks where possible.  I see your point though.

still lgtm.